src/BaseTable.tsx

```diff
@@ -249,12 +244,12 @@ class BaseTable extends React.PureComponent<IBaseTableProps, IBaseTableState> {
     return <TableRow {...rowProps} />;
   }
 
-  public renderRowCell = ({ isScrolling, columns, column, columnIndex, rowData, rowIndex, expandIcon }:
-    ICollectionObject) => {
+  public renderRowCell = ({ isScrolling, columns, column, columnIndex, rowData, rowIndex, expandIcon } : 
+    IRenderRowCellParam<T>) => {
     if (column[ColumnManager.PlaceholderKey]) {
       return (
         <div
-          key={`row-${rowData[this.props.rowKey]}-cell-${column.key}-placeholder`}
+          key={`row-${rowData[this.props.rowKey as number]}-cell-${column.key}-placeholder`}
```

Comment:
removing that casting brings back the ts complain.
```
Element implicitly has an 'any' type because index expression is not of type 'number'.ts(7015)
```
This seems obvious because I typed, `rowKey: string | number;` Is it always a number?

Reply:
Looks like, sometimes it is string. because, 
> BaseTable.tsx line 57
```
  private _flattenOnKeys: (tree: any[], keys: string[], dataKey: string) => any[];
```

And in `line 565`, this function is called with the third argument, `this.props.rowKey`. So, `rowKey` has to be string here.